content/blog/virtuous-circles.md

```diff
@@ -0,0 +1,92 @@
+to help with data imports, in a context where messy, inconsistent data from disparate sources often needs to be imported 
+into our databases. The code to perform these imports was formerly written from scratch as and when required, with no 
+reference to the structure of previous imports, and tended to be monolithic, difficult to review and untested. Dettl ("disinfect 
+your workflow!") encourages a cleaner approach by providing methods for a staged approach, separated into Extract, 
+Transform and Load steps, each of which incorporate running a suite of tests, and where each step must succeed (pass all tests)
+ before the next can be attempted. Users of the package need put no effort into writing code with a good structure, or 
+ remembering to write a full test suite. 
+
+[Orderly](https://vimc.github.io/orderly/), another RESIDE project, provides an infrastructure for producing reports or 
+analyses by running code against specified inputs and collecting outputs. It shares this basic collection of code and 
+data with the publishing tools described above. However, Orderly is not focussed on publication, and offers additional 
+reporting features such as the ability to introduce dependencies between reports by using the outputs of one report as 
+inputs to another, and the automatic versioning of reports every time they are run. Like rrtools, good practice in 
+reproducible results is built into the structure of the Orderly package, and like reproducible articles, further value 
+in given to users in the form of the Orderly Web front end which provides clarity and transparency, through visibility 
+of reports, their inputs, outputs and versioning information, and administrator features for assigning report-level permissions. 
+The users viewing the reports are often not the authors of the report code, but may be reviewers or other stakeholders.  
+
+All these tools incorporate beneficial practices into their basic structure, providing a virtuous template for achieving 
+clarity, completeness and separation of concerns, without requiring users to think about how to achieve these things 
+themselves. Further incentives to ensure continued use of the the tools can be provided through automation of tedious tasks, 
+and features to make review and engagement easier, both for primary users and those in other roles.
+
+But is there a danger that these tools are too prescriptive - that they might impose an idea of 'best practice' on users 
+which may be contentious, incomplete, outdated or wrong? Might they give a false impression of a fixed best practice, or 
+even discourage innovation?
+
+If there really is such an authoritarian risk, perhaps it is mitigated by democratising the software. Making these tools 
+modular - so users need only employ those elements which work for them; open - so they can be modified if required; 
+and interoperable with other tools and formats in the domain, could all help. 
+

```

Comment:
Had to read this a couple of times to get the grammar - perhaps the last few words should be "; these points could all help"

Reply:
Actually, it's probably fine!